### PR TITLE
Automated build changes

### PR DESCRIPTION
**Pull request by nerzhul**
* drop a useless ubuntu non lts version
* update gitlab ci syntax to modern one
* add debian 10 step on the gitlab ci pipeline
* switch docker build to Debian 10
* add .gradle folder to gitignore

## How to test

`docker build . ` for the docker part
See https://gitlab.com/nerzhul/minetest/-/jobs/504663970 for an example pipeline with debian:10

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -74,54 +74,78 @@ build:debian-8:
...
+ image: debian:10
+ before_script:
+   - apt-get update -y
+   - apt-get -y install build-essential libirrlicht-dev cmake libbz2-dev libpng-dev libjpeg-dev libxxf86vm-dev libgl1-mesa-dev libsqlite3-dev libogg-dev libvorbis-dev libopenal-dev libcurl4-gnutls-dev libfreetype6-dev zlib1g-dev libgmp-dev libjsoncpp-dev
+
+package:debian-10:
+  extends: .debpkg_template
+  image: debian:10
+  dependencies:
+    - build:debian-9
```
Comment (sfan5) [inline .gitlab-ci.yml:130]: `debian-10`?
same on line 138

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -133,102 +157,78 @@ build:ubuntu-14.04:
...
 #    LEVELDB_PKG: libleveldb1v5
-#  <<: *debpkg_template
 
 #deploy:ubuntu-16.10:
+#  extends: .debpkg_install
 #  image: ubuntu:yakkety
 #  dependencies:
 #    - package:ubuntu-16.10
 #  variables:
 #    LEVELDB_PKG: libleveldb1v5
```
Comment (sfan5) [inline .gitlab-ci.yml:223]: this is all also commented out, just remove it

File: .gitlab-ci.yml (hunk 1 above)
Comment (nerzhul) [reply .gitlab-ci.yml:130]: exact ! i will fix it

File: .gitlab-ci.yml (hunk 2 above)
Comment (nerzhul) [reply .gitlab-ci.yml:223]: yep good idea. We should rebuild up to date ubuntu images at a point

**Comment by nerzhul** [comment]
@sfan5 solved

**Pull request merged**